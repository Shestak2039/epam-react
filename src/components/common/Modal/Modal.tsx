import React, { useEffect } from 'react';
import { createPortal } from 'react-dom';

const modalRoot = document.getElementById( 'modal' );

export interface ModalProps {
    onClose: () => void;
}

const Modal: React.FunctionComponent<ModalProps> = ({children, onClose}) => {
    const element = document.createElement('div');

    useEffect(() => {
        if (modalRoot) {
            modalRoot.appendChild(element);
        }

        return () => {
            if (modalRoot) {
                modalRoot.removeChild(element);
            }
        }
    });

    return createPortal(
        (<div className="modal-background">
            <div className="modal-content">
                {children}
                <div className="modal-close" onClick={onClose} />
            </div>
        </div>)
        , element);
};

export default Modal;
