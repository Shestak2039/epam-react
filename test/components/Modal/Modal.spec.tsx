import React from 'react';
import { render } from '@testing-library/react';
import Modal from '../../../src/components/common/Modal/Modal';


describe('Modal', () => {
    test('renders Modal snapshot', () => {
        const mockCloseModal = jest.fn();
        
        const { asFragment } = render(<Modal onClose={mockCloseModal}/>);

        expect(asFragment()).toMatchSnapshot();
    });
});
