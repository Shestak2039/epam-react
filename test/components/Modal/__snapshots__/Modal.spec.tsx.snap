// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal renders Modal snapshot 1`] = `<DocumentFragment />`;
